### PR TITLE
Added new refactoring changes to dates and postal codes

### PR DESCRIPTION
**Pull request by sumhungyee**
Dates changed to now accept Strings in a "yyyy-MM-dd" format and times are no longer stored. As with previously, only valid dates are accepted.
Prefix "e/" in the CLI is now changed to "d/" to accommodate the "date" attribute. Error messages for invalid dates and ages are updated.
edit postal
- user can add with an S or no S.
- user can find postal codes with S or no S at the beginning of their sequence of numbers
- further update User guide to be updated with postal code changes

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -115,15 +117,17 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
...
-  e.g. `Hans Bo` will return `Hans Gruber`, `Bo Yang`
+  e.g. `Han Bo 101` will return `Abrahans Gruber`, `Boeing Yang` and all persons whose postal codes begin with `101`.
 
 Examples:
-* `find John` returns `john` and `John Doe`
-* `find alex david` returns `Alex Yeoh`, `David Li`<br>
-  ![result for 'find alex david'](images/findAlexDavidResult.png)
+* `find Boe` returns `Boeing` and `Wong Boe`
+* `find alex david 101` returns `Alexander Peterson`, `Allison Tan` (postal code), `Davidson Li`<br>
+  ![result for 'find alex david 101'](..%2F..%2F..%2FDownloads%2Fimage%20%281%29.png)
```
Comment (markusyeo) [inline docs/UserGuide.md:130]: Is this a placeholder image? if not could you place it into `docs/images/`

File: docs/UserGuide.md [hunk 2]
```diff
@@ -180,10 +184,10 @@ _Details coming soon ..._
 
 Action | Format, Examples
 --------|------------------
-**Add** | `add n/NAME p/PHONE_NUMBER e/DATE a/AGE [t/TAG]…​` <br> e.g., `add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+**Add** | `add n/NAME p/POSTAL_CODE d/DATE a/AGE [t/TAG]…​` <br> e.g., `add n/James Ho p/S222244 d/2000-11-11 a/123, t/URGENT`
```
Comment (markusyeo) [inline docs/UserGuide.md:187]: @valerietanhx to add userguide on multiple ways to add a user i.e. S222244 or 222244 is accepted

File: src/main/java/seedu/dengue/model/person/Date.java [hunk 3]
```diff
@@ -2,34 +2,29 @@
 
 import static java.util.Objects.requireNonNull;
 import static seedu.dengue.commons.util.AppUtil.checkArgument;
+import java.time.LocalDate;
```
Comment (markusyeo) [inline src/main/java/seedu/dengue/model/person/Date.java:5]: Add a newline above here

File: src/main/java/seedu/dengue/model/person/Date.java [hunk 4]
```diff
@@ -2,34 +2,29 @@
...
-    // alphanumeric and special characters
-    private static final String ALPHANUMERIC_NO_UNDERSCORE = "[^\\W_]+"; // alphanumeric characters except underscore
-    private static final String LOCAL_PART_REGEX = "^" + ALPHANUMERIC_NO_UNDERSCORE + "([" + SPECIAL_CHARACTERS + "]"
-            + ALPHANUMERIC_NO_UNDERSCORE + ")*";
-    private static final String DOMAIN_PART_REGEX = ALPHANUMERIC_NO_UNDERSCORE
-            + "(-" + ALPHANUMERIC_NO_UNDERSCORE + ")*";
-    private static final String DOMAIN_LAST_PART_REGEX = "(" + DOMAIN_PART_REGEX + "){2,}$"; // At least two chars
-    private static final String DOMAIN_REGEX = "(" + DOMAIN_PART_REGEX + "\\.)*" + DOMAIN_LAST_PART_REGEX;
-    public static final String VALIDATION_REGEX = LOCAL_PART_REGEX + "@" + DOMAIN_REGEX;
+            + "1. Date to be formatted as yyyy-MM-dd in 3 blocks separated by a hyphen '-'\n"
```
Comment (markusyeo) [inline src/main/java/seedu/dengue/model/person/Date.java:19]: For ease of use let's also accept yyyy/MM/dd

File: src/main/java/seedu/dengue/model/person/Date.java [hunk 5]
```diff
@@ -2,34 +2,29 @@
...
 /**
  * Represents a Person's dengue case date in the Dengue Hotspot Tracker.
  * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
  */
 public class Date {
 
-    private static final String SPECIAL_CHARACTERS = "+_.-";
-    public static final String MESSAGE_CONSTRAINTS = "Emails should be of the format local-part@domain "
+
+    public static final String MESSAGE_CONSTRAINTS = "Dates should be of the format yyyy-mm-dd"
```
Comment (markusyeo) [inline src/main/java/seedu/dengue/model/person/Date.java:17]: Should this be yyyy-MM-dd?

File: src/main/java/seedu/dengue/model/person/Postal.java [hunk 6]
```diff
@@ -21,9 +21,15 @@ public class Postal {
      * @param postal A valid postal number.
      */
     public Postal(String postal) {
+        boolean hasStart = postal.substring(0, 1).equals("S");
         requireNonNull(postal);
         checkArgument(isValidPostal(postal), MESSAGE_CONSTRAINTS);
-        value = postal;
+        if (hasStart) {
```
Comment (markusyeo) [inline src/main/java/seedu/dengue/model/person/Postal.java:27]: This should be on a different level of abstraction. Also are we storing with the S or without? I was thinking stripping to the integer value and then we display as such:
`S(<number>)`

File: src/main/java/seedu/dengue/model/util/SampleDataUtil.java [hunk 7]
```diff
@@ -19,23 +19,23 @@
 public class SampleDataUtil {
     public static Person[] getSamplePersons() {
         return new Person[] {
-            new Person(new Name("Alex Yeoh"), new Postal("674388"), new Date("alexyeoh@example.com"),
-                new Age("Blk 30 Geylang Street 29, #06-40"),
+            new Person(new Name("Alex Yeoh"), new Postal("674388"), new Date("2000-11-11 00:00"),
```
Comment (markusyeo) [inline src/main/java/seedu/dengue/model/util/SampleDataUtil.java:22]: Update DT format to match the updated Date format

File: docs/UserGuide.md (hunk 2 above)
Comment (sumhungyee) [reply docs/UserGuide.md:187]: small letter s works as well

File: src/main/java/seedu/dengue/model/person/Date.java (hunk 5 above)
Comment (sumhungyee) [reply src/main/java/seedu/dengue/model/person/Date.java:17]: right

File: src/main/java/seedu/dengue/model/person/Postal.java (hunk 6 above)
Comment (sumhungyee) [reply src/main/java/seedu/dengue/model/person/Postal.java:27]: seems right

**Pull request closed without merging**